kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/OpenShiftStartupStrategy.java

```diff
@@ -0,0 +1,198 @@
+                        ContainerManager containerManager,
+                        KieServerState currentState,
+                        AtomicBoolean kieServerActive) {
+
+        String kieServerId = currentState.getConfiguration()
+                                         .getConfigItem(KieServerConstants.KIE_SERVER_ID).getValue();
+
+        try (OpenShiftClient client = (OpenShiftClient) createAdaptiveClient()) {
+
+            Set<KieContainerResource> containers = prepareContainers(currentState.getContainers());
+            containerManager.installContainersSync(kieServer, containers, currentState, new KieServerSetup());
+
+            new Thread(new WatchRunner(kieServerId)).start();
+
+            triggerRollout(client, kieServerId, true);
+        }
+    }
+
+    private static boolean triggerRollout(OpenShiftClient client, String kieServerId, boolean completed) {
+
+        boolean pullTrigger = false;
+        String triggerName = KIE_SERVER_ROLLOUT_IN_PROGRESS + "-" + kieServerId;
+
+        if (completed) {
+            // Cleanup the configmap related to rollout-in-progress
+            pullTrigger = client.configMaps()
+                                .withName(triggerName).delete();
+            if (pullTrigger)
+                logger.info("Kie server: {}, rollout - End", KIE_SERVER_INSTANCE_ID);
+        } else {
+            ConfigMap cm = client.configMaps().withName(kieServerId).get();
+            Map<String, String> ann = cm.getMetadata().getAnnotations();
+
+            if (ann != null && ann.containsKey(KieServerStateCloudRepository.ROLLOUT_REQUIRED)) {
+                // Create temporary rollout-in-progress configmap
+                try {
+                    client.configMaps().create(new ConfigMapBuilder()
+                                                                     .withNewMetadata()
+                                                                     .withName(triggerName)
+                                                                     .withLabels(Collections.singletonMap(KIE_SERVER_INSTANCE_ID, kieServerId))
+                                                                     .endMetadata().build());
+
+                    // Cleanup the configmap annotation related to rollout-in-progress
+                    ann.remove(KieServerStateCloudRepository.ROLLOUT_REQUIRED);
+                    client.configMaps().createOrReplace(cm);
+
+                    pullTrigger = true;
+
+                    logger.info("Kie server: {}, rollout - Begin", KIE_SERVER_INSTANCE_ID);
+                } catch (KubernetesClientException kce) {
+                    logger.info("Rollout is in progress. {}", kce.getMessage());
+                }
+            }
+        }
+
+        return pullTrigger;
+    }
+
+    @Override
+    public String toString() {
+        return "OpenShiftStartupStrategy - " + "deploys only kie containers defined from OpenShift ConfigMap, " + "ignores kie containers given by controller";
```

Comment:
nitpick: These strings can be merged into the one.

Reply:
Committed the change as per suggested.